packages/core/src/payment/strategies/stripe-upe/stripe-upe-payment-strategy.ts

```diff
@@ -124,14 +124,23 @@ export default class StripeUPEPaymentStrategy implements PaymentStrategy {
             return await this._executeWithAPM(payment.methodId);
         }
 
-        const { paymentIntent, error } = await this._stripeUPEClient.confirmPayment(this._mapStripePaymentData());
+        let result;
+        let catchedConfirmError = false;
+        const paymentMethod = this._store.getState().paymentMethods.getPaymentMethodOrThrow(methodId);
+        const stripePaymentData = this._mapStripePaymentData();
+
+        try {
+            result = await this._stripeUPEClient.confirmPayment(stripePaymentData);
+        } catch (error) {
+            catchedConfirmError = true;
+        }
 
-        if (error) {
-            this._throwDisplayableStripeError(error);
+        if (result?.error) {
+            this._throwDisplayableStripeError(result.error);
             throw new PaymentMethodFailedError();
```

Comment:
Do we still need this?

Reply:
Yes. If we receive an error when calling `confirmPayment`, we must throw the exception. Only in case we catch an exception when calling `confirmPayment` would we continue with the flow.